feature: 배송 등록·조회 API 및 추적 연동 추가

cje206 opened the pull request:
## #️⃣ 연관된 이슈

> #160 

## #️⃣ 작업 내용

> - delivery-service 모듈 신규 생성 및 설정 반영
> - 배송 엔티티/리포지토리 포트-어댑터 구조 구성
> - 배송 등록·단건 조회 API와 요청/응답 DTO 신규 생성
> - 배송 추적 배치 스케줄러 신규 생성 및 실행 흐름 구성

## #️⃣ 테스트 결과

> 배송 등록/조회 가능
> 5분 단위로 배치 실행
> 10분 단위로 배송 추적(배송 완료/취소 제외)

## #️⃣ 변경 사항 체크리스트

- [x] 코드에 영향이 있는 모든 부분에 대한 테스트를 작성하고 실행했나요?
- [ ] 문서를 작성하거나 수정했나요? (필요한 경우)
- [x] 코드 컨벤션에 따라 코드를 작성했나요?
- [ ] 본 PR에서 발생할 수 있는 모든 의존성 문제가 해결되었나요?

## #️⃣ 스크린샷 (선택)

> 관련된 스크린샷이 있다면 여기에 첨부해주세요.

## #️⃣ 리뷰 요구사항 (선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요. 
> 예시: 이 부분의 코드가 잘 작동하는지 테스트해 주실 수 있나요?

## 📎 참고 자료 (선택)

> 관련 문서, 스크린샷, 또는 예시 등이 있다면 여기에 첨부해주세요

cje206 committed:
feature: 배송 등록·조회 API 및 추적 연동 추가

- delivery-service 모듈 신규 생성 및 설정 반영
- 배송 엔티티/리포지토리 포트-어댑터 구조 구성
- 배송 등록·단건 조회 API와 요청/응답 DTO 신규 생성
- 배송 추적 배치 스케줄러 신규 생성 및 실행 흐름 구성

metroz1 commented on modi/delivery-service/build.gradle:
```diff
@@ -0,0 +1,7 @@
+    // 공통 모듈 공유
+    implementation project(':common')
+
+    implementation 'org.springframework.boot:spring-boot-starter-webflux'
```
지금 현재 공통 라이브러리에 대한 부분들은 루트에 있는 build.gradle에서 통합으로 관리하고 있습니다.  여기서 common 가져오지 않아도 될 것 같아요!

metroz1 commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/application/dto/DeliveryCreateResponse.java:
```diff
@@ -0,0 +1,12 @@
+import com.coc.modi.delivery.delivery.domain.DeliveryStatus;
+
+public record DeliveryCreateResponse(
+		Long id,
```
요전에도 말씀 드렸던 것처럼 response에 단순하게 id가 아닌 deliveryId 이런 식으로 하는게 어떨까요??

metroz1 commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/application/dto/DeliveryDetailResponse.java:
```diff
@@ -0,0 +1,31 @@
+import java.time.LocalDateTime;
+
+public record DeliveryDetailResponse(
+		Long id,
```
여기도요!

metroz1 reviewed (changes requested):
리뷰 사항 참고해서 수정 혹은 코멘트 부탁드립니다

kdkrkwhr commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/scheduler/DeliveryTrackingService.java:
```diff
@@ -0,0 +1,63 @@
+				
+			} catch (Exception e) {
+				
+				d.markTrackedNow();
```
현재 실패해도 위 method에서 업데이트가 되는거 같은데 이렇게 처리 된 이유가 있을까요?

kdkrkwhr commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/presentation/DeliveryController.java:
```diff
@@ -0,0 +1,44 @@
+	
+	// 배송 등록
+	@PostMapping
+	public ResponseEntity<ApiResponse<DeliveryCreateResponse>> createDelivery(@Valid @RequestBody DeliveryCreateRequest request) {
```
현재 배송에 대한 등록 API에 대한 인증을 추가하면 어떨지 의견 드립니다!
현재는 누구나 배송에 대한 등록이 될 것 같습니다!

kdkrkwhr commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/scheduler/DeliveryTrackingJob.java:
```diff
@@ -0,0 +1,29 @@
+	
+	private final DeliveryTrackingService deliveryTrackingService;
+	
+	@Scheduled(fixedDelayString = "PT5M")
```
혹시 요 설정값을 하드코딩하지 않고 yml로 분리시키는건 어떨지 의견 드려봅니다!
차후 운영을하며 유지보수 시 용이할 것 같습니다!

kdkrkwhr replied:
그리고 이건 제안 사항인데요!
현재는 인스턴스 한 대 단독으로 운영을 하게 될 테지만 차후 여러 인스턴스로 구성이 될 경우를 대비해 스케줄 사용 시, 중복 실행되지 않도록하는 부분을 고민하여 추가해보면 어떨지도 의견 드립니다!

kdkrkwhr commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/infrastructure/CarrierTrackingClient.java:
```diff
@@ -0,0 +1,119 @@
+					.bodyToMono(TrackerDeliveryResponse.class)
+					.timeout(Duration.ofSeconds(10))
+					.block();
+		} catch (Exception e) {
```
현재 모든 예외에 대해 일시적 오류로 판단을 하는 것 같은데 이를 분기시켜보면 어떨까요?
예외 타입에 따라 처리 방식이 바뀌진 않을까 합니다!

kdkrkwhr commented on modi/delivery-service/src/main/java/com/coc/modi/delivery/delivery/application/DeliveryStatusMapper.java:
```diff
@@ -0,0 +1,27 @@
+@Component
+public class DeliveryStatusMapper {
+	
+	public DeliveryStatus map(TrackingResult result) {
```
혹시 위 meyyhod 호출 시 result 가 null로 들어올 수 있진 않을까요?
간단한 체크를 추가해보면 어떨지 의견 드려봅니다!

cje206 committed:
feature: 코드 리뷰 수정 사항 반영

- 단일 인스턴스 환경뿐 아니라 다중 인스턴스 구성 시에도
  스케줄이 중복 실행되지 않도록 방지 로직 추가
- 스케줄 관련 설정을 yml로 분리하여 관리
- 결과값 null 체크 및 예외 분기 보완

metroz1 reviewed (approved):
확인 했습니다
